src/navigation/reroute.js

```diff
@@ -49,12 +49,16 @@ export function reroute(pendingPromises = [], eventArguments) {
         wasNoOp = false;
       }
 
-      return Promise.all(loadPromises)
-        .then(finishUpAndReturn)
-        .catch(err => {
-          callAllEventListeners();
-          throw err;
-        });
+      return (
+        Promise.all(loadPromises)
+          .then(callAllEventListeners)
```

Comment:
`finishUpAndReturn` used to do five things:

1. Fire single-spa routing events
2. Reset `appChangeUnderway`
3. Call `callEventListeners`
4. Move on to next promise queued up by appChangeUnderway
5. Set return value to list of mounted apps

Before singleSpa.start() is called, we only need 3 and 5, and we don't want 1, 2, and 4. So I switched to doing 3 and 5 right here inline.